### ISSUE TITLE
Regression in Multi-curl Asynchronous Calls after Upgrading Curl Version (8.9.1 → 8.11.0)

### ISSUE THREAD

**Issue by wilfredNJH** [curl/curl#15639]
### I did this

I recently upgraded my project’s version of libcurl from **8.9.1** to **8.11.0** and encountered a regression in the behavior of multi-curl asynchronous calls. The same issue persists with version **8.10.0** as well. I’m building libcurl with **ENABLE_SCHANNEL=yes**, relying on Windows WinSSL/Schannel for SSL/TLS handling.

Interestingly, this issue only appears on older Windows versions. When I upgraded to a newer version of Windows, the issue seems to be resolved.

**Usage:**
TimerCallbackFunc uses - boost::asio::deadline_timer mTimer; 
```c
    curl_multi_setopt(mpMulti, CURLMOPT_SOCKETFUNCTION, &SocketCallbackFunc); 
    curl_multi_setopt(mpMulti, CURLMOPT_SOCKETDATA, this);
    
    curl_multi_setopt(mpMulti, CURLMOPT_TIMERFUNCTION, &TimerCallbackFunc);
    curl_multi_setopt(mpMulti, CURLMOPT_TIMERDATA, this);
    
    curl_multi_setopt(mpMulti, CURLMOPT_MAX_HOST_CONNECTIONS, 5);
```

**Key Details**:
- The issue manifests when making asynchronous multi-curl requests.
- The regression happens specifically with older Windows versions and not on newer ones.
- The upgrade from 8.9.1 to 8.11.0 is the first version where this issue occurs (also affecting 8.10.0).
- I’ve checked the change logs and haven’t identified any relevant changes that might explain this behavior.

Was there any change introduced in libcurl 8.10.0 or 8.11.0 that could potentially result in this regression, specifically with Windows Schannel SSL/TLS handling or multi-curl asynchronous behavior?

### I expected the following

There should be no issues with the usage of multi-curl with SChannel SSL/TLS, regardless of the Windows version. The regression observed on older Windows versions is unexpected.

### curl/libcurl version

**On the Windows 10 machine that does not work** 
Microsoft Windows [Version 10.0.17763.1217]
curl 7.55.1 (Windows) libcurl/7.55.1 WinSSL
Release-Date: [unreleased]
Protocols: dict file ftp ftps http https imap imaps pop3 pop3s smtp smtps telnet tftp
Features: AsynchDNS IPv6 Largefile SSPI Kerberos SPNEGO NTLM SSL

**On the Windows 10 machine that works** 
Microsoft Windows [Version 10.0.19044.4529]
curl 8.7.1 (Windows) libcurl/8.7.1 Schannel zlib/1.3 WinIDN
Release-Date: 2024-03-27
Protocols: dict file ftp ftps http https imap imaps ipfs ipns mqtt pop3 pop3s smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS HSTS HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM SPNEGO SSL SSPI threadsafe Unicode UnixSockets

### operating system

Microsoft Windows [Version 10.0.17763.1217] & Microsoft Windows [Version 10.0.19044.4529]

**Comment by jay** [comment]
What regression are you experiencing? Check out #15154 and #15627. There was some issue with CURLMOPT_TIMERFUNCTION. I suspect that the latter probably solves the former but we're waiting to hear from the reporter.

**Comment by wilfredNJH** [comment]
Hey @jay, #15627 did indeed resolve the regression I'm seeing. Thank you! 

Any idea why a Windows upgrade might seem to fix/alleviate the regression seen?

**Comment by jay** [comment]
> Will this fix be a part of the future 8.11.1 release and is there a timeline for the next release?

Yes, the next release is on December 11.

> Any idea why a Windows upgrade might seem to fix/alleviate the regression seen?

Red herring probably. Differences in the network.
